### PR TITLE
feat: Create Timezone Endpoint

### LINKED ISSUE #233: [Feat] Implement Create Timezone Endpoint
## Description

This issue involves creating an endpoint to create a new timezone in the system.

## Endpoint

**Method:** POST  
**URL:** /api/v1/timezones

## Request Body

**Content-Type:** application/json  
**Body:**
```json
{
  "timezone": "America/New_York",
  "gmtOffset": "-05:00",
  "description": "Eastern Standard Time"
}
```

## Success Response

**Status Code:** 201
**Body:**
```json
{
  "id": "a1b2c3d4-e5f6-g7h8-i9j0-k1l2m3n4o5p",
  "timezone": "America/New_York",
  "gmtOffset": "-05:00",
  "description": "Eastern Standard Time",
  "status_code": 201,
  "message": "Timezone created successfully"
}
```

## Error Responses

- **Conflict (409)**  
  **Body:**
  ```json
{
  "status_code": 409,
  "error": "Timezone 'America/New_York' already exists.",
  "message": "Timezone already exists"
}
  ```

- **Forbidden (403)**  
  **Body:**
  ```json
  {
    "status_code": 403,
    "message": "Forbidden"
  }
  ```

- **Bad Request (400)**  
  **Body:**
  ```json
  {
    "status_code": 400,
    "message": "Invalid timezone data"
  }
  ```

## Checklist

- [ ] Implement POST /api/v1/timezones endpoint.
- [ ] Ensure endpoint requires valid Authorization token.
- [ ] Validate the provided timezone.
- [ ] Create unit tests for the endpoint.
- [ ] Update API documentation.

### Error Handling

- Invalid request parameters (400 Bad Request)
- Timezone already exists (409 Conflict)
- Authentication failure (401 Unauthorized)
- Authorization failure (403 Forbidden)
- Internal server errors (500 Internal Server Error)

## Security Measures

1. Use HTTPS for all API communications.
2. Implement rate limiting to prevent abuse.
3. Validate and sanitize all input parameters.
4. Use environment variables for storing sensitive data.
5. Implement proper authentication and authorization checks.

## Tasks

1. **Create CreateTimezoneCommand and handler.**
2. **Implement controller action for the endpoint.**
3. **Implement authorization checks.**
4. **Implement error handling and logging.**
5. **Write unit and integration tests.**
6. **Add Swagger documentation for the API endpoint.**
7. **Perform security review and penetration testing.**
8. **Deploy to staging environment for testing.**

### PR DESCRIPTION
**Pull request by dorisjenny27**
# Description

Closes #233

Changes proposed
What were you told to do?
Create an endpoint to create a new timezone in the system.

Create Timezone Endpoint

What did you do?
Created an endpoint that enable users create timezone.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [ ] My code follows the code style of this project.
- [ ] This PR does not contain plagiarized content.
- [ ] The title and description of the PR is clear and explains the approach.
- [ ] I am making a pull request against the **dev branch** (left side).
- [ ] My commit messages styles matches our requested structure.
- [ ] My code additions will fail neither code linting checks nor unit test.
- [ ] I am only making changes to files I was requested to.

# Screenshots/Videos

### CODE REVIEW

File: src/Hng.Application/Features/Timezones/Commands/CreateTimezoneCommand.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using Hng.Application.Features.Timezones.Dtos;
+using MediatR;
+
+namespace Hng.Application.Features.Timezones.Commands
+{
+    public class CreateTimezoneCommand : IRequest<CreateTimezoneResponseDto>
+    {
+        public string Timezone { get; set; }
+        public string GmtOffset { get; set; }
+        public string Description { get; set; }
```
Comment (Godhanded) [inline src/Hng.Application/Features/Timezones/Commands/CreateTimezoneCommand.cs:10]: Include JsonPropertyName on all dtos for snake_case

File: src/Hng.Web/Controllers/LanguageController.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿using Hng.Application.Features.Languages.Dtos;
+using Hng.Application.Features.Languages.Queries;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Hng.Web.Controllers
+{
+    [Route("api/[controller]")]
```
Comment (Godhanded) [inline src/Hng.Web/Controllers/LanguageController.cs:8]: Please confirm if controller name matches specification in nestjs docs

**Review by Godhanded — CHANGES_REQUESTED** [review]

File: src/Hng.Web/Controllers/LanguageController.cs (hunk 2 above)
Comment (dorisjenny27) [reply src/Hng.Web/Controllers/LanguageController.cs:8]: Yes, it matches nestjs docs specification

File: src/Hng.Application/Features/Timezones/Commands/CreateTimezoneCommand.cs (hunk 1 above)
Comment (dorisjenny27) [reply src/Hng.Application/Features/Timezones/Commands/CreateTimezoneCommand.cs]: Done

**Review by Godhanded — APPROVED** [review]

**Pull request merged**